src/client.py

```diff
@@ -0,0 +1,99 @@
+from http.client import *
+import socket
+import errno
+import sys
+
+_GLOBAL_DEFAULT_TIMEOUT = object()
+
+def _create_connection(address, timeout=_GLOBAL_DEFAULT_TIMEOUT, source_address=None, *, all_errors=False, proto=0):
+    host, port = address
+    exceptions = []
+    for res in socket.getaddrinfo(host, port, 0, socket.SOCK_STREAM):
+        af, socktype, _, canonname, sa = res
+        sock = None
+        try:
+            sock = socket.socket(af, socktype, proto)
+            # if timeout is not _GLOBAL_DEFAULT_TIMEOUT:
+            #     sock.settimeout(timeout)
+            if source_address:
+                sock.bind(source_address)
+            sock.connect(sa)
+            # Break explicitly a reference cycle
+            exceptions.clear()
+            return sock
+
+        except Exception as exc:
+            if not all_errors:
+                exceptions.clear()  # raise only the last error
+            exceptions.append(exc)
+            if sock is not None:
+                sock.close()
+
+    if len(exceptions):
+        try:
+            if not all_errors:
+                raise exceptions[0]
+            raise ExceptionGroup("create_connection failed", exceptions)
+        finally:
+            # Break explicitly a reference cycle
+            exceptions.clear()
+    else:
+        raise "getaddrinfo returns an empty list"
+
+class HTTPConnection(http.client.HTTPConnection):
+    def __init__(self, host, port=None, timeout=socket._GLOBAL_DEFAULT_TIMEOUT,
+                 source_address=None, blocksize=8192):
+        super().__init__(host, port, timeout, source_address, blocksize)
```

Comment:
can you not set `self._create_connection = _create_connection` and that's it, no need to redifine `connect`, no?